src/internal/small_vec.rs

```diff
@@ -128,6 +136,40 @@ impl<'de, T: serde::Deserialize<'de>> serde::Deserialize<'de> for SmallVec<T> {
     }
 }
 
+impl<T> IntoIterator for SmallVec<T> {
```

Comment:
I'm not clear why we need this. If it's only used in nonperformance critical code then `iter().cloned()` is probably easier to maintain.
But if it is critical, then this looks correct.

Reply:
I added this to ensure that during deserialization there is no extra allocation required. In your example code a `Vec` was allocated but this can now be done with a stack allocated `SmallVec`. The `IntoIter` trait for `SmallVec` can then be used to construct the `Range` without allocation too. I thought this was nice, but Im ok with removing it. WDYT?

Reply:
I have not particularly considered the performance of (de)serialization as important, as it's primarily used for creating test files. That being said, no reason not to make it better, this is a step in the right direction. The deserialization for `SmallVec` itself does an extra allocation, but we can always fix that later.

Reply:
Cool and thanks for pointing that out. I also fixed the serialization of SmallVec. 😄

Reply:
Thanks! Looks good!